http/logical.go

```diff
@@ -170,7 +170,9 @@ func respondLogical(w http.ResponseWriter, r *http.Request, path string, dataOnl
 				},
 			}
 		} else {
-			httpResp = logical.SanitizeResponse(resp)
+			sanitizedHttp := logical.SanitizeResponse(resp)
```

Comment:
You can simply use `httpResp` here and then set the request ID that comes out, rather than creating a new variable and assigning afterwards.